libs/designer/src/lib/ui/panel/panelRoot.tsx

```diff
@@ -0,0 +1,84 @@
+import type { AppDispatch } from '../../core';
+import { useIsDarkMode } from '../../core/state/designerOptions/designerOptionsSelectors';
+import { useCurrentPanelModePanelMode, useIsPanelCollapsed } from '../../core/state/panel/panelSelectors';
+import { clearPanel } from '../../core/state/panel/panelSlice';
+import { ErrorPanel } from './errorsPanel/errorsPanel';
+import { NodeDetailsPanel } from './nodeDetailsPanel/nodeDetailsPanel';
+import { usePanelTabs } from './nodeDetailsPanel/tabInitialization';
+import { NodeSearchPanel } from './nodeSearchPanel/nodeSearchPanel';
+import { RecommendationPanelContext } from './recommendation/recommendationPanelContext';
+import { WorkflowParametersPanel } from './workflowParametersPanel/workflowParametersPanel';
+import { Panel, PanelType } from '@fluentui/react';
+import type { CommonPanelProps } from '@microsoft/designer-ui';
+import { PanelLocation, PanelSize } from '@microsoft/designer-ui';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useDispatch } from 'react-redux';
+
+export interface PanelRootProps {
+  panelLocation?: PanelLocation;
+  displayRuntimeInfo: boolean;
+}
+
+const layerProps = {
+  hostId: 'msla-layer-host',
+  eventBubblingEnabled: true,
+};
+
+export const PanelRoot = (props: PanelRootProps): JSX.Element => {
+  const { panelLocation, displayRuntimeInfo } = props;
+  const dispatch = useDispatch<AppDispatch>();
+
+  const isDarkMode = useIsDarkMode();
+
+  const collapsed = useIsPanelCollapsed();
+  const currentPanelMode = useCurrentPanelModePanelMode();
+
+  usePanelTabs(); // This initializes tabs for the node details panel, can't be run twice so it lives here instead of in the panel
```

Comment:
```suggestion
  usePanelTabs(); // This function initializes the tabs for the node details panel. Can't run twice so the function exists here, rather than in the panel.
```